stock_partner_delivery_window/models/res_partner.py

```diff
@@ -49,19 +57,13 @@ def get_delivery_windows(self, day_name=None):
         :param day: The day name (see time.weekday, ex: 0,1,2,...)
         :return: dict partner_id: delivery_window recordset
         """
-        res = {}
+        self.ensure_one()
         domain = [("partner_id", "in", self.ids)]
```

Comment:
```suggestion
        domain = [("partner_id", "=", self.id)]
```

Reply:
Since we don't want to modify the behavior of existing methods, I reverted this change.